classes/data/Transfer.class.php

```diff
@@ -227,7 +227,7 @@ public static function fromGuestsOf($user) {
         // Gather user's guests transfers
         $transfers = array();
         foreach(Guest::fromUser($user) as $gv) {
-            $transfers = array_merge($transfers, $gv->transfers);
+	        $gv->transfers += $transfers;
```

Comment:
Does this work? The following test code didn't work for me:

	<?php
	$a = array('a');
	$b = array('b');
	$b += $a;
	var_export($b);

Yields

	array (
	  0 => 'b',
	)

Reply:
I notice the same thing. There is a subtle difference between using += and array_merge() which might not be what we are expecting in filesender in general.

If you have identical keys in two arrays of length 3 then array_merge will preserve all elements and change the keys to work. += will retain only the key-values from one array and ignore the other. It seems there is less possible surprise using array_merge() especially when the indexes 

```
php >  $a = array(1,2,3);
php > $b = array(10,11,12);
php > $c = array_merge($a,$b);
php > var_dump($c);
array(6) {
  [0]=>
  int(1)
  [1]=>
  int(2)
  [2]=>
  int(3)
  [3]=>
  int(10)
  [4]=>
  int(11)
  [5]=>
  int(12)
}
php > $d = $a;
php > $d += $b;
php > var_dump($d);
array(3) {
  [0]=>
  int(1)
  [1]=>
  int(2)
  [2]=>
  int(3)
}
```

In this case $gv->transfers is from Transfer::fromGuest() which is from self::all() which uses the primary key as the array key. So, without having tested it I think it might be ok. Though still might not be worth it to have to verify each time instead of just using array_merge() which does what one would expect.